### PR TITLE
feat: expose number of rows covered per delta

### PR DESCRIPTION
**Pull request by chebbyChefNEQ**
This PR adds a field to `index_stats` where we now return the number of rows covered by each delta in `num_indexed_rows_per_delta`

TODO:
- [ ] tests

### CODE REVIEW

**Review by rpgreen — APPROVED** [review]

File: rust/lance/src/index.rs [hunk 1]
```diff
@@ -550,6 +555,7 @@ impl DatasetIndexExt for Dataset {
             "num_indexed_rows": num_indexed_rows,
             "num_unindexed_fragments": num_unindexed_fragments,
             "num_unindexed_rows": num_unindexed_rows,
+            "num_indexed_rows_per_delta": num_indexed_rows_per_delta,
```
Comment (wjones127) [inline rust/lance/src/index.rs:558]: Is this implicitly in the same order that `list_indices` returns? Do we even guarantee an order there?

**Review by wjones127 — COMMENTED** [review]
I'm a little concerned about whether the result is useful if we don't even guarantee the order of the index deltas. Otherwise seems fine. Though I feel like it's not that hard for users to compute this themselves, as the `fragment_bitmap` and `num_rows` should both be exposed publicly, in Python and Rust.

File: rust/lance/src/index.rs (hunk 1 above)
Comment (chebbyChefNEQ) [reply rust/lance/src/index.rs:558]: I think we guarantee append order, let me check

**Comment by chebbyChefNEQ** [comment]
> I'm a little concerned about whether the result is useful if we don't even guarantee the order of the index deltas. Otherwise seems fine. Though I feel like it's not that hard for users to compute this themselves, as the `fragment_bitmap` and `num_rows` should both be exposed publicly, in Python and Rust.

We don't expose the per delta fragment bitmap to python or rust. This is only a stop gap so we have something for calculating the "base" index rows. Eventually this should be something we can track natively in v3

**Comment by chebbyChefNEQ** [comment]
> > I'm a little concerned about whether the result is useful if we don't even guarantee the order of the index deltas. Otherwise seems fine. Though I feel like it's not that hard for users to compute this themselves, as the `fragment_bitmap` and `num_rows` should both be exposed publicly, in Python and Rust.
> 
> We don't expose the per delta fragment bitmap to python or rust. This is only a stop gap so we have something for calculating the "base" index rows. Eventually this should be something we can track natively in v3

Actually, looks like it's a `pub trait` let me see if this is exposed thru lancedb

**Comment by chebbyChefNEQ** [comment]
going to merge this for the time being as this saves us from drifting from OSS.

**Pull request merged**